week2/index.js

```diff
@@ -4,14 +4,56 @@ var options = process.argv.slice(2);
 
 var command = options[0];
 
+
 switch (command) {
   case 'help':
-  default:
-    showHelp();
+    help();
     break;
   case 'list':
     listTodos();
     break;
+  case 'add':
+    add()
+    break;
+  case 'remove':
+    remove()
+    break;
+  case 'reset':
+    reset()
+    break;
+  case 'update':
+    update()
+    break;
+  default:
+    showHelp();
+}
+
+function help() {
+  console.log(`
+options :
+1- help : show help section.
+2- add : add a todo item.
+3- list: show current todo's, or show an appropriate text if there are no todos.
+4- remove: remove a todo item by its 1-base myapp.js.
+5- reset: remove all todo items from the list.
+`);
+};
+
+//find number after instruction
+function num() {
+  let number = parseInt(options[1])
+
+  //avoid type anything but numbers
+  if ((typeof number === 'number' && !isNaN(number))) {
+    return number
+
+  } else if (options[1] == null) {
```

Comment:
Careful with this `options[1] == null` comparison. While it does work right now, the actual value of an out-of-bounds array element is undefined, not null. So `options[1] === null` would not do what you want, whereas `options[1] === undefined` does, and is more likely to be what you intended for this line to mean.